### DISCUSSION TITLE
Summary on RequestManager Button

### DISCUSSION THREAD

**Discussion by hrj** [UprootLabs/gngr#273] [Dev Talk]
@behrens-web  Let's continue the discussion here.

> I see, Does every task have to get accepted/rejected inside RequestManager?

Yes!

> If so, does it happen inside isRequestPermitted?

I think so, but not very sure of the exact details. I have been more focussed on the layout engine recently. This part of the code was written many years back :)

> (I wanna make sure we're on the same page here.)

I think you are on the right track. Please feel free to ask questions. Whenever I get a chance, will try to dive into more details.

**Comment by behrens-web** [comment]
In that case, I will create rudimentary counters for accepted and rejected
requests and add the statistics to the buttonPane within RequestManager
because I dont see anywhere else to put it.

**Reply by hrj** [reply]
Yeah sure, go ahead! This is just to let the user see a quick overview of the requests.

If you want to go further, you can also indicate the percentage of rejected requests somehow. For example, as a bar-chart or pie-chart (miniature versions that will fit within the button icon).

Have you used uMatrix anytime? It was a chrome/firefox plugin that showed a red/green map on the plugin icon, indicating the percentage of rejected/accepted requests.

**Comment by behrens-web** [comment]
I'm afraid I'm not familiar with it. The only plugin I have intimate
knowledge of is DESMO-J. I can display the number of accepted and rejected
requests as is or as a percentage, but that's kind of all I can do.

**Comment by behrens-web** [comment]
When I press the ReqMgr button and it brings up a series of green and red
cells, does each cell represent a request? Does each row represent a
request? How do I know how many requests there are?

**Comment by behrens-web** [comment]
Alright, I think I got it. On the left side of the "OK" button it displays
the number of accepted requests. On the right side of the "OK" button it
displays the number of total requests.

**Comment by behrens-web** [comment]
Ok, I've submitted a pull request with my contribution to RequestManager to
gngr. Fair warning, I made the first commit before I knew how to use GPG,
but I have another commit with a GPG key attached to it because I couldn't
figure out how append worked.

**Reply by hrj** [reply]
No worries. And I am thrilled to see your contribution.

I hope you will make a fresh PR with the changes I suggested.
